miner/src/service_transaction_checker.rs

```diff
@@ -44,9 +55,42 @@ impl ServiceTransactionChecker {
 
 	/// Checks if given address is whitelisted to send service transactions.
 	pub fn check_address<C: CallContract + RegistryInfo>(&self, client: &C, sender: Address) -> Result<bool, String> {
+		trace!(target: "txqueue", "Checking service transaction checker contract from {}", sender);
+		if let Some(allowed) = self.certified_addresses_cache.try_read().as_ref().and_then(|c| c.get(&sender)){
```

Comment:
`.as_ref()` seems redundant, also missing space before `{`

Reply:
I cannot do this without `.as_ref()` since  the `c` doesn't live long enough in this case. I tried to do it with `.map`, but in this case `Option<Option<&bool>>` is returned, so one more check is needed. Not sure what would be better solution here.